### PR TITLE
Add API to star and unstar challenge.

### LINKED ISSUE #1183: Add API to star a challenge.

### PR DESCRIPTION
**Pull request by RishabhJain2018**
fixes #1183 @deshraj Please review the PR.

### CODE REVIEW

File: apps/challenges/serializers.py [hunk 1]
```diff
@@ -151,3 +152,27 @@ class Meta:
...
+                kwargs['data']['challenge'] = challenge.pk
+            request = context.get('request')
+            if request:
+                kwargs['data']['user'] = request.user.pk
+
+    class Meta:
+        model = StarChallenge
+        fields = ('user', 'challenge', 'count_stars',)
+
+    def get_count_stars(self, obj):
```
Comment (deshraj) [inline apps/challenges/serializers.py]: Change the name with `get_count`

File: apps/challenges/urls.py [hunk 2]
```diff
@@ -34,5 +34,8 @@
     url(r'(?P<challenge_pk>[0-9]+)/phase/(?P<challenge_phase_pk>[0-9]+)'
         r'/download_all_submissions/(?P<file_type>[A-Za-z]+)/$',
         views.download_all_submissions,
-        name='download_all_submissions')
+        name='download_all_submissions'),
+    url(r'(?P<challenge_pk>[0-9]+)/star_challenge/$',
```
Comment (deshraj) [inline apps/challenges/urls.py]: Change `star_challenge` with `star`

File: apps/challenges/views.py [hunk 3]
```diff
@@ -835,3 +836,34 @@ def download_all_submissions(request, challenge_pk, challenge_phase_pk, file_typ
...
+    a challenge.
+    """
+    challenge = get_challenge_model(challenge_pk)
+
+    try:
+        starred_challenge = StarChallenge.objects.get(user=request.user,
+                                                      challenge=challenge)
+        starred_challenge.delete()
+        serializer = StarChallengeSerializer(starred_challenge)
+        response_data = {'success': 'Challenge successfully unstarred!',
```
Comment (deshraj) [inline apps/challenges/views.py]: Change the message with `Challenge unstarred successfully!`

File: apps/challenges/views.py [hunk 4]
```diff
@@ -835,3 +836,34 @@ def download_all_submissions(request, challenge_pk, challenge_phase_pk, file_typ
...
+    """
+    API endpoint for starring and unstarring
+    a challenge.
+    """
+    challenge = get_challenge_model(challenge_pk)
+
+    try:
+        starred_challenge = StarChallenge.objects.get(user=request.user,
+                                                      challenge=challenge)
+        starred_challenge.delete()
```
Comment (deshraj) [inline apps/challenges/views.py]: Please add a support for the boolean field here.

**Comment by RishabhJain2018** [comment]
@deshraj Please review the PR.

File: apps/challenges/views.py [hunk 5]
```diff
@@ -840,3 +841,48 @@ def download_all_submissions(request, challenge_pk, challenge_phase_pk, file_typ
...
+    if request.method == 'POST':
+        try:
+            starred_challenge = StarChallenge.objects.get(user=request.user,
+                                                          challenge=challenge)
+            starred_challenge.is_starred = not starred_challenge.is_starred
+            starred_challenge.save()
+            serializer = StarChallengeSerializer(starred_challenge)
+            response_data = serializer.data
+            return Response(response_data, status=status.HTTP_200_OK)
+        except:
```
Comment (deshraj) [inline apps/challenges/views.py]: Please add a specific exception here.

**Pull request merged**